### DISCUSSION TITLE
is there a plugin available for parsing swift-style block directives?

### DISCUSSION THREAD

**Discussion by sowelipililimute** [remarkjs/remark#1317] [Q&A] [answered]
documentation of the swift format: https://apple.github.io/swift-markdown/documentation/markdown/blockdirective

TLDR
```swift
@DirectiveName(key: value) {
  hi i am some content
  @MoreStuff
  @YetMore {
  }
}
```

**Comment by wooorm** [comment, marked as answer]
Not that I am aware of.

To do that correctly, you’d need to integrate with micromark, the markdown parser. See its docs on extensions.

But, I would recommend using directives or MDX instead. Both rather popular. It might not work with Swift tools, but it’ll work with more tools.
